src/pytorch_lightning/strategies/tpu_spawn.py

```diff
@@ -77,6 +77,18 @@ def __init__(
         self.tpu_global_core_rank = 0
         self.start_method = "fork"
 
+    @property
+    def global_rank(self) -> int:
+        return self.tpu_global_core_rank
+
+    @property
+    def local_rank(self) -> int:
+        return self.tpu_local_core_rank
+
+    @property
+    def world_size(self) -> int:
+        return xm.xrt_world_size()
```

Comment:
should it use xla cluster env?

Reply:
With TPUs, you only get the right ranks with the XLA api inside the processes, whereas we are calling it before hand.
 So the local and global ranks are always set to 0. 
 Dealing withe XLA cluster env would be a follow up PR.

Reply:
that as well